src/darwin/Framework/CHIPTests/MTRSwiftPairingTests.swift

```diff
@@ -0,0 +1,88 @@
+import Matter
+import XCTest
+
+// This more or less parallels the "no delegate" case in MTRPairingTests
+
+struct Constants {
+    static let localPort = 5541
+    static let vendorID = 0xFFF1
+    static let onboardingPayload = "MT:Y.K90SO527JA0648G00"
+    static let deviceID = 0x12344321
+    static let timeoutInSeconds : UInt16 = 3
+}
+
+class MTRSwiftPairingTestControllerDelegate : NSObject, MTRDeviceControllerDelegate {
+    let expectation: XCTestExpectation
+
+    init(withExpectation providedExpectation: XCTestExpectation) {
+        expectation = providedExpectation
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, statusUpdate status: MTRCommissioningStatus) {
+        XCTAssertNotEqual(status, MTRCommissioningStatus.failed)
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, commissioningSessionEstablishmentDone error: Error?) {
+        XCTAssertNil(error)
+
+        do {
+            try controller.commissionNode(withID: Constants.deviceID as NSNumber, commissioningParams: MTRCommissioningParameters())
+        } catch {
+            XCTFail("Could not start commissioning of node")
+        }
+
+        // Keep waiting for commissioningComplete
+    }
+
+    func controller(_ controller: MTRDeviceController, commissioningComplete error: Error?, nodeID: NSNumber?) {
+        XCTAssertNil(error)
+        XCTAssertEqual(nodeID, Constants.deviceID as NSNumber)
+        expectation.fulfill()
+    }
+}
+
+class MTRSwiftPairingTests : XCTestCase {
+    @available(macOS, introduced: 13.3)
+    func test001_BasicPairing() throws {
+        let factory = MTRDeviceControllerFactory.sharedInstance()
+        XCTAssertNotNil(factory)
+
+        let storage = MTRTestStorage()
+        let factoryParams = MTRDeviceControllerFactoryParams(storage: storage)
+        factoryParams.port = Constants.localPort as NSNumber
+
+        try factory.start(factoryParams)
+        XCTAssertTrue(factory.isRunning)
+
+        let testKeys = MTRTestKeys()
+
+        let params = MTRDeviceControllerStartupParams(ipk: testKeys.ipk, fabricID: 1, nocSigner: testKeys)
+        params.vendorID = Constants.vendorID as NSNumber;
+
+        let controller = try factory.createController(onNewFabric: params)
```

Comment:
Ok, so the use of try here indicates that createController(...) might throw an error (and in several other places in the code). So the bare try here will just propagate the error, which might be what you want idk. But if you want to add some error decoration you could use a do catch block, as you have elsewhere. Another good option here would be try?, which would make controller an optional, and then you could test using guard let or XCTUnwrap. It can start getting verbose though, and if I was writing more than one test I'd be hiding a lot of this setup behind common helper functions or in setUp() (or a combo of both probably) so we could be verbose with the error messages but still keep the test as concise as possible.

Reply:
Yes, this will need refactoring if/when we add more than one test, absolutely.

Reply:
I'll do `do` with XCTFail for nicer error messages.